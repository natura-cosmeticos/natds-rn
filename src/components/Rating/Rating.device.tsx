import { storiesOf } from '@storybook/react-native';
import {
  Counter, ReadOnly, Interactive, Input,
} from './Rating.stories';

storiesOf('Rating', module)
  .add('Counter', Counter)
  .add('ReadOnly', ReadOnly)
  .add('Input', Input)
  .add('Interactive', Interactive);
