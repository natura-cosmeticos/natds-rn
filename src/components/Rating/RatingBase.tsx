import React from 'react';

import { withTheme } from 'styled-components/native';
import { Size } from '@naturacosmeticos/natds-themes/react-native';
import { IconButtonBase } from '../IconButton/IconButtonBase';
import { Icon } from '../Icon/Icon';
import { Theme } from '../../common/themeSelectors';

export interface RatingBaseProps {
  testID?: string
  onPress?: () => void
  disabled?: boolean
  size: keyof Size
  accessibilityHint?: string
  accessibilityLabel?: string
  iconActive?: boolean
  iconFilled?: boolean
  isClickable?: boolean
  theme: Theme
}

const RatingBaseComponent = ({
  testID, disabled = false, onPress, size, theme,
  accessibilityHint, accessibilityLabel, iconFilled = true, iconActive = true, isClickable = true,
}: RatingBaseProps): JSX.Element => {
  const getRatingColor = ({ color }: Theme) => (iconFilled && iconActive && !disabled ? '#F8B546' : color.mediumEmphasis);

  const iconName = iconFilled ? 'filled-action-rating' : 'outlined-action-rating';

  return (
    <IconButtonBase
      testID={testID}
      disabled={disabled || !isClickable}
      onPress={onPress}
      size={size}
      IconComponent={(
        <Icon
          accessibilityHint={accessibilityHint}
          accessibilityLabel={accessibilityLabel}
          accessibilityRole="imagebutton"
          size={size}
          name={iconName}
          style={{ color: getRatingColor(theme) }}
          testID={`${testID}-icon`}
        />
      )}
    />
  );
};

export const RatingBase = withTheme(RatingBaseComponent);
