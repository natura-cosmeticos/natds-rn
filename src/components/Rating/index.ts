export * from './Rating';
export * from './Rating.types';
