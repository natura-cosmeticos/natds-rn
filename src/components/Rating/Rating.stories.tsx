import React, { useState } from 'react';
import { boolean, text, select } from '@storybook/addon-knobs';
import { StoryContainer } from '../../common/HelperComponents/StoryContainer';
import { RatingValueProps } from './Rating.types';

import { Rating } from './Rating';

const description = () => `
- - -

  ### NOTE:
  This component is available in the following variants:

  - ✅ Counter
  - ✅ Input
  - ✅ Read-Only

With the following attribute status:

  - ✅ Hint (available on input and counter variants)
  - **Size:**
      - ✅ \`Small\` (available on input and read-only variants)
      - ✅ \`Standard\` (available on input and read-only variants)
      - ✅ \`Semi\` (available in all variants)
      - ✅ \`SemiX\` (available in all variants)
      - ✅ \`Medium\` (available only in input variant)
  - **Alignment** (available only in counter variant)
    - ✅ \`left\`
    - ✅ \`right\`
  - ✅ **Disabled** (available only in input variant)
  - ✅ **Rate** (available on input and read-only variants)
  - **Interaction state**:
      - ✅ \`Enabled\`
      - ✅ \`Press\`

- - -
`;

export default {
  component: Rating,
  parameters: {
    componentSubtitle: 'Rating subtitle',
    docs: {
      extractComponentDescription: description,
    },
  },
  title: 'Components|Rating',
};

const variants = {
  counter: 'counter',
  input: 'input',
  readOnly: 'read-only',
};

const sizes = {
  medium: 'medium',
  semi: 'semi',
  semix: 'semiX',
  small: 'small',
  standard: 'standard',
};

const rates = {
  0: 0,
  1: 1,
  2: 2,
  3: 3,
  4: 4,
  5: 5,
};


export const Input = () => {
  const [rating, setRating] = useState<RatingValueProps>(0);

  return (
    <StoryContainer title='Input'>
      <Rating variant='input' rate={rating} label='Placeholder' size='semi' onPress={(value => setRating(value))}/>
    </StoryContainer>
  );
};

export const ReadOnly = () => (
  <StoryContainer title='Read-Only'>
    <Rating variant='read-only' rate={3} size='semi' />
  </StoryContainer>
);

export const Counter = () => (
  <StoryContainer title='Counter'>
    <Rating variant='counter' label='Placeholder' size='semi' />
  </StoryContainer>
);

export const Interactive = () => (
  <StoryContainer title='Interactive'>
    <Rating
      variant={select('Variant', variants, 'read-only') as any}
      label={text('Label', 'Placeholder')}
      size={select('Size', sizes, 'semi') as any}
      align={select('Align', ['left', 'right'], 'left') as any}
      disabled={boolean('Disabled', false)}
      rate={select('Rate', rates, 1) as RatingValueProps}
      onPress={(() => {})}
    />
  </StoryContainer>
);
