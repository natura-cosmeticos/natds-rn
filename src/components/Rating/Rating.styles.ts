/* eslint-disable max-len */
import styled from 'styled-components/native';
import { CSSObject } from 'styled-components';
import { Theme } from '@naturacosmeticos/natds-themes/react-native';
import { RatingCounterProps, RatingProps } from './Rating.types';

type VariantStyleProps = {
  theme: Theme
} & Pick<RatingProps, 'variant'>

type AlignStyleProps = {
  theme: Theme
} & Pick<RatingCounterProps, 'align'>


const getLabelMargin = ({ theme, variant }: VariantStyleProps) => (
  variant === 'input' && {
    marginTop: theme.spacing.micro,
  }
);

const getWrapperAlign = ({ variant }: Pick<RatingProps, 'variant'>) => (
  variant === 'counter' && {
    alignItems: 'center',
  }
);

const getWrapperDirection = ({ variant, align }: AlignStyleProps & VariantStyleProps) => {
  if (variant === 'input') {
    return 'column';
  }

  if (align === 'right') {
    return 'row-reverse';
  }

  return 'row';
};

const getContentMargin = ({ theme, align, variant }: VariantStyleProps & AlignStyleProps) => {
  if (align === 'right' && variant === 'counter') {
    return {
      marginLeft: theme.spacing.micro,
    };
  }
  if (align === 'left' && variant === 'counter') {
    return {
      marginRight: theme.spacing.micro,
    };
  }

  return { marginHorizontal: 0 };
};

export const Wrapper = styled.View(({ theme, variant, align = 'left' }: VariantStyleProps & AlignStyleProps): CSSObject => ({
  ...getWrapperAlign({ variant }),
  flexDirection: getWrapperDirection({ align, theme, variant }),
}));

export const Content = styled.View(({ theme, variant, align }: VariantStyleProps & AlignStyleProps): CSSObject => ({
  ...getContentMargin({ align, theme, variant }),
  flexDirection: 'row',
}));

export const Label = styled.Text(({ theme, variant }: VariantStyleProps): CSSObject => ({
  ...getLabelMargin({ theme, variant }),
  color: theme.color.mediumEmphasis,
  fontFamily: theme.rating[variant].label.primary.fontFamily,
  fontSize: theme.rating[variant].label.fontSize,
  fontWeight: theme.rating[variant].label.primary.fontWeight,
  letterSpacing: theme.rating[variant].label.letterSpacing,
  lineHeight: theme.rating[variant].label.fontSize * theme.rating[variant].label.lineHeight,
}));
