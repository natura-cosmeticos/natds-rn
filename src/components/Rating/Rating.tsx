/* eslint-disable max-len */
import React from 'react';

import { Wrapper, Content, Label } from './Rating.styles';
import {
  RatingCounterProps, RatingInputProps, RatingProps, RatingReadOnlyProps,
} from './Rating.types';
import { RatingBase } from './RatingBase';

export const isRatingInput = (props: RatingProps): props is RatingInputProps => props.variant === 'input';
export const isRatingCounter = (props: RatingProps): props is RatingCounterProps => props.variant === 'counter';
export const isRatingReadOnly = (props: RatingProps): props is RatingReadOnlyProps => props.variant === 'read-only';

export const renderTimes = (props: RatingProps) => (isRatingCounter(props) ? 1 : 5);
export const isFilled = (value: number, props: RatingProps) => !isRatingInput(props) || (props.rate && value + 1 <= props.rate);
export const isActive = (value: number, props: RatingProps) => (!isRatingCounter(props) && (props.rate && value + 1 <= props.rate)) || isRatingCounter(props);

export const isDisabled = (props: RatingProps) => isRatingInput(props) && props.disabled;

export const Rating = (props: RatingProps) => {
  const {
    size = 'semi', variant, testID = 'ds-rating', ...rest
  } = props;

  return (
    <Wrapper testID={testID} variant={variant} {...rest}>
      <Content testID={`${testID}-content`} variant={variant} {...rest}>
        {Array(renderTimes(props)).fill(0).map((_, index) => (
          (isRatingCounter(props) || isRatingReadOnly(props) || isRatingInput(props)) && (
            <RatingBase
              testID={`${testID}-${index + 1}`}
              key={index.toString()}
              size={size}
              onPress={isRatingInput(props) ? () => props.onPress(index + 1) : undefined}
              iconActive={!!isActive(index, props)}
              iconFilled={!!isFilled(index, props)}
              disabled={isDisabled(props)}
              isClickable={isRatingInput(props)}
            />
          )
        ))}
      </Content>
      {(isRatingInput(props) || isRatingCounter(props)) && (
        <Label testID={`${testID}-label`} variant={variant}>{props.label}</Label>
      )}
    </Wrapper>
  );
};
