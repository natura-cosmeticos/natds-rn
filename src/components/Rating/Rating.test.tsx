import React from 'react';
import { fireEvent } from '@testing-library/react-native';

import { renderWithTheme } from '../../../test/testHelpers';
import { Rating } from './Rating';
import { RatingBase } from './RatingBase';
import {
  isActive, isDisabled, isFilled, renderTimes,
} from '.';
import {
  RatingCounterProps, RatingInputProps, RatingReadOnlyProps,
} from './Rating.types';

jest.mock('../Icon/Icon');
jest.mock('../../common/themeSelectors');

describe('Rating Variants', () => {
  describe('Counter Variant', () => {
    it('should render with counter variant', () => {
      const { getByTestId, toJSON, getAllByTestId } = renderWithTheme(<Rating variant='counter' label='placeholder' />);

      expect(getAllByTestId(/-icon/).length).toBe(1);
      expect(getByTestId('ds-rating')?.props).toHaveProperty('variant', 'counter');
      expect(getByTestId('ds-rating')).toHaveStyle({ alignItems: 'center', flexDirection: 'row' });
      expect(toJSON()).toMatchSnapshot();
    });

    it('should render with label aligned on right', () => {
      const { getByTestId } = renderWithTheme(<Rating variant='counter' align='right' label='placeholder' />);

      expect(getByTestId('ds-rating')).toHaveStyle({ alignItems: 'center', flexDirection: 'row-reverse' });
    });

    it('should render with label aligned on left', () => {
      const { getByTestId } = renderWithTheme(<Rating variant='counter' align='left' label='placeholder' />);

      expect(getByTestId('ds-rating-content')).toHaveStyle({ marginRight: 4 });
    });

    it('should render with another size', () => {
      const { getByTestId } = renderWithTheme(<Rating variant='counter' size="standard" label='placeholder' />);

      expect(getByTestId('ds-rating-1-icon')?.props).toHaveProperty('size', 'standard');
    });
  });


  describe('Input Variant', () => {
    it('should render with input variant', () => {
      const { getByTestId, toJSON, getAllByTestId } = renderWithTheme(<Rating variant='input' label='placeholder' onPress={() => {}} />);

      expect(getAllByTestId(/-icon/).length).toBe(5);
      expect(getByTestId('ds-rating')?.props).toHaveProperty('variant', 'input');
      expect(getByTestId('ds-rating')).toHaveStyle({ flexDirection: 'column' });
      expect(getByTestId('ds-rating-label')).toBeTruthy();
      expect(toJSON()).toMatchSnapshot();
    });

    it('should call onPress when item is pressed', () => {
      const onPressMock = jest.fn();

      const { getByTestId } = renderWithTheme(<Rating variant='input' label='placeholder' onPress={onPressMock} />);

      fireEvent.press(getByTestId('ds-rating-1-button'));

      expect(onPressMock).toHaveBeenCalled();
    });

    it('should render one yellow start when rate is equal one', () => {
      const { getByTestId } = renderWithTheme(<Rating testID='rating' variant='input' rate={1} label='placeholder' onPress={() => {}} />);

      expect(getByTestId('rating-1-icon')).toHaveStyle({ color: '#F8B546' });
    });

    it('should render label when variant is input', () => {
      const { getByTestId } = renderWithTheme(<Rating variant='input' label='placeholder' onPress={() => {}} />);

      expect(getByTestId('ds-rating-label')).toBeTruthy();
    });

    it('should not call onPress when disabled', () => {
      const onPressMock = jest.fn();

      const { getByTestId } = renderWithTheme(<Rating variant='input' disabled label='placeholder' onPress={onPressMock} />);

      fireEvent.press(getByTestId('ds-rating-1-button'));

      expect(onPressMock).not.toHaveBeenCalled();
    });
  });


  describe('ReadOnly Variant', () => {
    it('should render with read-only variant', () => {
      const { toJSON, getByTestId } = renderWithTheme(<Rating variant='read-only' />);

      expect(getByTestId('ds-rating')?.props).toHaveProperty('variant', 'read-only');
      expect(toJSON()).toMatchSnapshot();
    });

    it('should render three yellow stars and two gray starts', () => {
      const { getByTestId } = renderWithTheme(<Rating rate={3} variant='read-only' />);

      expect(getByTestId('ds-rating-3-icon')).toHaveStyle({ color: '#F8B546' });
      expect(getByTestId('ds-rating-4-icon')).toHaveStyle({ color: '#777777' });
    });
  });

  describe('RatingBase component', () => {
    it('should render a base component with medium size', () => {
      const { toJSON } = renderWithTheme(<RatingBase size='medium' />);

      expect(toJSON()).toMatchSnapshot();
    });

    it('should not call onPress when isclickable props is false', () => {
      const onPressMock = jest.fn();

      const { getByTestId } = renderWithTheme(<RatingBase testID="base" size='medium' isClickable={false} onPress={onPressMock}/>);

      fireEvent.press(getByTestId('base-button'));

      expect(onPressMock).not.toHaveBeenCalled();
    });

    it('should not call onPress when disabled', () => {
      const onPressMock = jest.fn();

      const { getByTestId } = renderWithTheme(<RatingBase testID="base" size='medium' disabled onPress={onPressMock}/>);

      fireEvent.press(getByTestId('base-button'));

      expect(onPressMock).not.toHaveBeenCalled();
    });

    it('should not render a filled item in component row', () => {
      const { getByTestId } = renderWithTheme(<RatingBase testID="base" size='medium' iconFilled={false} />);

      expect(getByTestId('base-icon')?.props).toHaveProperty('name', 'outlined-action-rating');
    });

    it('should not render a active item in component row', () => {
      const { getByTestId } = renderWithTheme(<RatingBase testID="base" size='medium' iconActive={false} />);

      expect(getByTestId('base-icon')?.props).toHaveProperty('name', 'filled-action-rating');
      expect(getByTestId('base-icon')).toHaveStyle({ color: '#777777' });
    });
  });

  describe('Rating external methods', () => {
    it('should return only one item in row when variant is counter', () => {
      const defaultProps: RatingCounterProps = {
        label: 'placeholder',
        variant: 'counter',
      };

      const result = renderTimes(defaultProps);

      expect(result).toBe(1);
    });

    it('should return five itens in row when variant is different than counter', () => {
      const props: RatingInputProps = {
        onPress: () => {},
        variant: 'input',
      };

      const result = renderTimes(props);

      expect(result).toBe(5);
    });

    it('should return true when variant is different than input', () => {
      const props: RatingCounterProps = {
        label: 'placeholder',
        variant: 'counter',
      };

      const result = isFilled(0, props);

      expect(result).toBe(true);
    });

    it('should return true when variant is input with rate equal or higher than item index', () => {
      const props: RatingInputProps = {
        onPress: () => {},
        rate: 2,
        variant: 'input',
      };

      const result = isFilled(0, props);

      expect(result).toBe(true);
    });

    it('should return disabled prop value when variant is input', () => {
      const props: RatingInputProps = {
        disabled: false,
        onPress: () => {},
        rate: 2,
        variant: 'input',
      };

      const result = isDisabled(props);

      expect(result).toBe(false);
    });

    it('should return active equal true when variant is counter', () => {
      const props: RatingCounterProps = {
        label: 'placeholder',
        variant: 'counter',
      };

      const result = isActive(0, props);

      expect(result).toBe(true);
    });

    it('should return active true when variant is different than counter with rate equal or higher than item index', () => {
      const props: RatingReadOnlyProps = {
        rate: 1,
        variant: 'read-only',
      };

      const result = isActive(0, props);

      expect(result).toBe(true);
    });
  });
});
