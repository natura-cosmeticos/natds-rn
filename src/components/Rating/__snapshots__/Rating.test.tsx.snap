// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating Variants Counter Variant should render with counter variant 1`] = `
<View
  label="placeholder"
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
  testID="ds-rating"
  variant="counter"
>
  <View
    label="placeholder"
    style={
      Array [
        Object {
          "flexDirection": "row",
          "marginHorizontal": 0,
        },
      ]
    }
    testID="ds-rating-content"
    variant="counter"
  >
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 50,
        }
      }
      testID="ds-rating-1-button"
    >
      <View
        backgroundStyle="none"
        disabled={true}
        size="semi"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 50,
              "height": "px",
              "justifyContent": "center",
              "width": "px",
            },
            Object {},
          ]
        }
        testID="ds-rating-1-background"
      >
        <Text
          accessibilityRole="imagebutton"
          name="filled-action-rating"
          size="semi"
          style={
            Object {
              "color": "#F8B546",
            }
          }
          testID="ds-rating-1-icon"
        >
          filled-action-rating
        </Text>
      </View>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#777777",
          "fontFamily": "Roboto",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": 0.22,
          "lineHeight": 21,
        },
      ]
    }
    testID="ds-rating-label"
    variant="counter"
  >
    placeholder
  </Text>
</View>
`;

exports[`Rating Variants Input Variant should render with input variant 1`] = `
<View
  label="placeholder"
  onPress={[Function]}
  style={
    Array [
      Object {
        "flexDirection": "column",
      },
    ]
  }
  testID="ds-rating"
  variant="input"
>
  <View
    label="placeholder"
    onPress={[Function]}
    style={
      Array [
        Object {
          "flexDirection": "row",
          "marginHorizontal": 0,
        },
      ]
    }
    testID="ds-rating-content"
    variant="input"
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 50,
        }
      }
      testID="ds-rating-1-button"
    >
      <View
        backgroundStyle="none"
        disabled={false}
        size="semi"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 50,
              "height": "px",
              "justifyContent": "center",
              "width": "px",
            },
            Object {},
          ]
        }
        testID="ds-rating-1-background"
      >
        <Text
          accessibilityRole="imagebutton"
          name="outlined-action-rating"
          size="semi"
          style={
            Object {
              "color": "#777777",
            }
          }
          testID="ds-rating-1-icon"
        >
          outlined-action-rating
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 50,
        }
      }
      testID="ds-rating-2-button"
    >
      <View
        backgroundStyle="none"
        disabled={false}
        size="semi"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 50,
              "height": "px",
              "justifyContent": "center",
              "width": "px",
            },
            Object {},
          ]
        }
        testID="ds-rating-2-background"
      >
        <Text
          accessibilityRole="imagebutton"
          name="outlined-action-rating"
          size="semi"
          style={
            Object {
              "color": "#777777",
            }
          }
          testID="ds-rating-2-icon"
        >
          outlined-action-rating
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 50,
        }
      }
      testID="ds-rating-3-button"
    >
      <View
        backgroundStyle="none"
        disabled={false}
        size="semi"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 50,
              "height": "px",
              "justifyContent": "center",
              "width": "px",
            },
            Object {},
          ]
        }
        testID="ds-rating-3-background"
      >
        <Text
          accessibilityRole="imagebutton"
          name="outlined-action-rating"
          size="semi"
          style={
            Object {
              "color": "#777777",
            }
          }
          testID="ds-rating-3-icon"
        >
          outlined-action-rating
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 50,
        }
      }
      testID="ds-rating-4-button"
    >
      <View
        backgroundStyle="none"
        disabled={false}
        size="semi"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 50,
              "height": "px",
              "justifyContent": "center",
              "width": "px",
            },
            Object {},
          ]
        }
        testID="ds-rating-4-background"
      >
        <Text
          accessibilityRole="imagebutton"
          name="outlined-action-rating"
          size="semi"
          style={
            Object {
              "color": "#777777",
            }
          }
          testID="ds-rating-4-icon"
        >
          outlined-action-rating
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 50,
        }
      }
      testID="ds-rating-5-button"
    >
      <View
        backgroundStyle="none"
        disabled={false}
        size="semi"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 50,
              "height": "px",
              "justifyContent": "center",
              "width": "px",
            },
            Object {},
          ]
        }
        testID="ds-rating-5-background"
      >
        <Text
          accessibilityRole="imagebutton"
          name="outlined-action-rating"
          size="semi"
          style={
            Object {
              "color": "#777777",
            }
          }
          testID="ds-rating-5-icon"
        >
          outlined-action-rating
        </Text>
      </View>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#777777",
          "fontFamily": "Roboto",
          "fontSize": 12,
          "fontWeight": "400",
          "letterSpacing": 0.38,
          "lineHeight": 18,
          "marginTop": 4,
        },
      ]
    }
    testID="ds-rating-label"
    variant="input"
  >
    placeholder
  </Text>
</View>
`;

exports[`Rating Variants RatingBase component should render a base component with medium size 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 50,
    }
  }
  testID="undefined-button"
>
  <View
    backgroundStyle="none"
    disabled={false}
    size="medium"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderRadius": 50,
          "height": "px",
          "justifyContent": "center",
          "width": "px",
        },
        Object {},
      ]
    }
    testID="undefined-background"
  >
    <Text
      accessibilityRole="imagebutton"
      name="filled-action-rating"
      size="medium"
      style={
        Object {
          "color": "#F8B546",
        }
      }
      testID="undefined-icon"
    >
      filled-action-rating
    </Text>
  </View>
</View>
`;

exports[`Rating Variants ReadOnly Variant should render with read-only variant 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
    ]
  }
  testID="ds-rating"
  variant="read-only"
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "marginHorizontal": 0,
        },
      ]
    }
    testID="ds-rating-content"
    variant="read-only"
  >
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 50,
        }
      }
      testID="ds-rating-1-button"
    >
      <View
        backgroundStyle="none"
        disabled={true}
        size="semi"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 50,
              "height": "px",
              "justifyContent": "center",
              "width": "px",
            },
            Object {},
          ]
        }
        testID="ds-rating-1-background"
      >
        <Text
          accessibilityRole="imagebutton"
          name="filled-action-rating"
          size="semi"
          style={
            Object {
              "color": "#777777",
            }
          }
          testID="ds-rating-1-icon"
        >
          filled-action-rating
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 50,
        }
      }
      testID="ds-rating-2-button"
    >
      <View
        backgroundStyle="none"
        disabled={true}
        size="semi"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 50,
              "height": "px",
              "justifyContent": "center",
              "width": "px",
            },
            Object {},
          ]
        }
        testID="ds-rating-2-background"
      >
        <Text
          accessibilityRole="imagebutton"
          name="filled-action-rating"
          size="semi"
          style={
            Object {
              "color": "#777777",
            }
          }
          testID="ds-rating-2-icon"
        >
          filled-action-rating
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 50,
        }
      }
      testID="ds-rating-3-button"
    >
      <View
        backgroundStyle="none"
        disabled={true}
        size="semi"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 50,
              "height": "px",
              "justifyContent": "center",
              "width": "px",
            },
            Object {},
          ]
        }
        testID="ds-rating-3-background"
      >
        <Text
          accessibilityRole="imagebutton"
          name="filled-action-rating"
          size="semi"
          style={
            Object {
              "color": "#777777",
            }
          }
          testID="ds-rating-3-icon"
        >
          filled-action-rating
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 50,
        }
      }
      testID="ds-rating-4-button"
    >
      <View
        backgroundStyle="none"
        disabled={true}
        size="semi"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 50,
              "height": "px",
              "justifyContent": "center",
              "width": "px",
            },
            Object {},
          ]
        }
        testID="ds-rating-4-background"
      >
        <Text
          accessibilityRole="imagebutton"
          name="filled-action-rating"
          size="semi"
          style={
            Object {
              "color": "#777777",
            }
          }
          testID="ds-rating-4-icon"
        >
          filled-action-rating
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 50,
        }
      }
      testID="ds-rating-5-button"
    >
      <View
        backgroundStyle="none"
        disabled={true}
        size="semi"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 50,
              "height": "px",
              "justifyContent": "center",
              "width": "px",
            },
            Object {},
          ]
        }
        testID="ds-rating-5-background"
      >
        <Text
          accessibilityRole="imagebutton"
          name="filled-action-rating"
          size="semi"
          style={
            Object {
              "color": "#777777",
            }
          }
          testID="ds-rating-5-icon"
        >
          filled-action-rating
        </Text>
      </View>
    </View>
  </View>
</View>
`;
